### PR TITLE
Add post categories

### PR DESCRIPTION
**Pull request by straight-shoota**
Categorizes blog posts into four categories:

* Project
* Technical
* Community
* Success stories (this is already started in #656, but a couple more are added here)

There a new index page for each of these categories showing all posts. They have the same design as the general blog index,
except for success stories, which keeps the design with post images from #656.
 
grafik

Every categorized post has the category in its breadcrumbs:

grafik

I also added some tags here and there but they're not used anywhere yet.

Goes on top of #656

### CODE REVIEW

**Comment by beta-ziliani** [comment]
Is there a reason for the tags to be lowercase (`project`, `technical`) instead of Capitalized?

**Comment by beta-ziliani** [comment]
Is it possible to, once you're in a category, to navigate between the articles of that category?

**Comment by beta-ziliani** [comment]
Forgot to mention: I LOVE the concept 🤩

File: _pages/used-in-production.md
Comment (beta-ziliani) [inline _pages/used-in-production.md:1]: what is this file?

File: _posts/2013-09-23-type-inference-part-1.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ title: Type inference (part 1)
 thumbnail: x=
 summary: How type inference works
 author: bcardiff
+categories: technical
+tags: language types
```
Comment (beta-ziliani) [inline _posts/2013-09-23-type-inference-part-1.md:7]: above they were `[listed, like, this]`, but here `like this`

File: _posts/2015-12-24-the-future-of-crystal.md [hunk 2]
```diff
@@ -3,6 +3,7 @@ title: The future of Crystal
 summary: A short story
 thumbnail: 🎄
 author: asterite
+categories: project
```
Comment (beta-ziliani) [inline _posts/2015-12-24-the-future-of-crystal.md:6]: missing tags

File: _posts/2016-06-28-split-the-repo.md [hunk 3]
```diff
@@ -3,6 +3,7 @@ title: Split the repo!
 summary: where the git...
 thumbnail: G
 author: bcardiff
+categories: project
```
Comment (beta-ziliani) [inline _posts/2016-06-28-split-the-repo.md:6]: missing tags

File: _posts/2016-12-26-insert-into-shard-values-crystal-db.md [hunk 4]
```diff
@@ -3,6 +3,9 @@ title: INSERT INTO shard VALUES ("crystal-db")
 summary: One db api to rule them all
 thumbnail: DB
 author: bcardiff
+categories: community
+tags: ecosystem
+tags: database
```
Comment (beta-ziliani) [inline _posts/2016-12-26-insert-into-shard-values-crystal-db.md:8]:
```suggestion
tags: ecosystem, database
```

File: _posts/2019-06-19-snapcraft-summit-montreal.md [hunk 5]
```diff
@@ -68,7 +71,7 @@ A basic `snapcraft.yaml` file to declare all the parts will look as follows:
 name: crystal-hello
 version: "1.0"
 summary: Create the hello snap
-description: Create the hello snap
+summary: Create the hello snap
```
Comment (beta-ziliani) [inline _posts/2019-06-19-snapcraft-summit-montreal.md:74]: This change should be reverted

File: _posts/2019-06-19-snapcraft-summit-montreal.md [hunk 6]
```diff
@@ -2,6 +2,9 @@
 title: Snapcraft Summit Montréal
 summary: Updates in Crystal distributions and other news
 author: bcardiff
+categories: community
+tags: events
+tags: packaging
```
Comment (beta-ziliani) [inline _posts/2019-06-19-snapcraft-summit-montreal.md:7]:
```suggestion
tags: events, packaging
```

File: _posts/2020-03-03-towards-crystal-1.0.md [hunk 7]
```diff
@@ -4,6 +4,7 @@ summary: The upcoming efforts to release Crystal 1.0
 thumbnail: +
 author: bcardiff
 comment_href: https://disqus.com/home/discussion/crystal-lang/towards_crystal_10_23/
+categories: project
```
Comment (beta-ziliani) [inline _posts/2020-03-03-towards-crystal-1.0.md:7]: missing tags

File: _posts/2020-08-20-preparing-our-shards-for-crystal-1.0.md [hunk 8]
```diff
@@ -4,6 +4,7 @@ summary: How we can prepare our shards for Crystal 1.0 or for upstream changes o
 thumbnail: +
 author: bcardiff
 comment_href: https://disqus.com/home/discussion/crystal-lang/preparing_our_shards_for_crystal_10/
+categories: technical
```
Comment (beta-ziliani) [inline _posts/2020-08-20-preparing-our-shards-for-crystal-1.0.md:7]: missing tags

**Review by beta-ziliani — CHANGES_REQUESTED** [review]
I have not finished, but I put enough to deserve a second iteration

**Comment by straight-shoota** [comment]
> Is there a reason for the tags to be lowercase (`project`, `technical`) instead of Capitalized?

No particular reason. We can capitalize them as well.

**Comment by straight-shoota** [comment]
> Is it possible to, once you're in a category, to navigate between the articles of that category?

I *think* this is not trivial in Jekyll. Also it's questionable wether you actually would want to do that. Having the chronology entail *all*  blog posts is also quite reasonable.

We could easily add a list of recent posts from the same category, for example.

File: _posts/2015-12-24-the-future-of-crystal.md (hunk 2 above)
Comment (straight-shoota) [reply _posts/2015-12-24-the-future-of-crystal.md:6]: Tags are just added as a byproduct and have no effect. I don't think it makes much sense to strive for completeness here.

**Comment by beta-ziliani** [comment]
> Having the chronology entail all blog posts is also quite reasonable.

Yes, unless you want to go and read _all_ of the post in a category. For instance, in the Home page we say "read more" for the success stories, and that takes you to the Success stories page, containing all the stories, and when you click one that takes you to a story linked to the rest of the posts. I would expect to be able to navigate the category instead.

But if it's not easy to do, no need to fret about it.

File: _sass/README.md
Comment (beta-ziliani) [inline _sass/README.md:1]: is this a legit file?

File: _samples/concurrency.cr
Comment (beta-ziliani) [inline _samples/concurrency.cr:1]: unrelated change

File: test/Ruby.txt
Comment (beta-ziliani) [inline test/Ruby.txt:1]: unrelated change?

File: test/Crystal.txt
Comment (beta-ziliani) [inline test/Crystal.txt:1]: unrelated change?

File: test.sh
Comment (beta-ziliani) [inline test.sh:1]: ??

File: _sass/frameworks/_pygment_dark.scss
Comment (beta-ziliani) [inline _sass/frameworks/_pygment_dark.scss:1]: is this relevant here?

File: _samples/bindings.cr
Comment (beta-ziliani) [inline _samples/bindings.cr:1]: unrelated change

File: _posts/2019-06-19-snapcraft-summit-montreal.md [hunk 9]
```diff
@@ -1,7 +1,9 @@
 ---
 title: Snapcraft Summit Montréal
-summary: Updates in Crystal distributions and other news
+description: Updates in Crystal distributions and other news
```
Comment (beta-ziliani) [inline _posts/2019-06-19-snapcraft-summit-montreal.md:3]: Is this change OK? I don't think `description` is used. Most post have`summary`, except for a few (and that's probably an error).

File: _posts/2019-06-19-snapcraft-summit-montreal.md (hunk 5 above)
Comment (beta-ziliani) [reply _posts/2019-06-19-snapcraft-summit-montreal.md:73]: I don't understand, this is marked as resolved but it wasn't.

**Review by beta-ziliani — CHANGES_REQUESTED** [review]
More spurious(?) files.

File: test.sh
Comment (straight-shoota) [reply test.sh:1]: I have no idea how these files got checked in 🤷

File: _sass/frameworks/_pygment_dark.scss
Comment (straight-shoota) [reply _sass/frameworks/_pygment_dark.scss:1]: No. It's a mistake.

File: _posts/2019-06-19-snapcraft-summit-montreal.md (hunk 9 above)
Comment (beta-ziliani) [reply _posts/2019-06-19-snapcraft-summit-montreal.md:3]: @straight-shoota ping

Comment (beta-ziliani) [reply _posts/2019-06-19-snapcraft-summit-montreal.md:3]: And the one below 👇

**Review by beta-ziliani — APPROVED** [review]
Good to go. However, I note that there's something funky about posts having `description` or `summary`. We should unify these two.

**Comment by straight-shoota** [comment]
Yes, I'm working on clarifying that.
